### PR TITLE
fix indexer model

### PR DESCRIPTION
**Pull request by scholtz**
current implementation failing in current testnet.. for example ClosingAmount may be larger than 2B

OptedOutAtRound failing to deserialize

### CODE REVIEW

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs [hunk 1]
```diff
@@ -783,7 +783,7 @@ public StateSchema(ulong? numUint = 0, ulong? numByteSlice = 0)
             // to ensure "numByteSlice" is required (not null)
             if (numByteSlice == null)
             {
-                throw new InvalidDataException("numByteSlice is a required property for StateSchema and cannot be null");
+                //throw new InvalidDataException("numByteSlice is a required property for StateSchema and cannot be null");
```
Comment (WesleyMiller1998) [inline dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:786]: This conditional is janky. Is there a reason to keep it that I'm not seeing?

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs [hunk 2]
```diff
@@ -792,7 +792,7 @@ public StateSchema(ulong? numUint = 0, ulong? numByteSlice = 0)
             // to ensure "numUint" is required (not null)
             if (numUint == null)
             {
-                throw new InvalidDataException("numUint is a required property for StateSchema and cannot be null");
+                //throw new InvalidDataException("numUint is a required property for StateSchema and cannot be null");
```
Comment (WesleyMiller1998) [inline dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:795]: Same with this conditional

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs [hunk 3]
```diff
@@ -583,15 +583,15 @@ public partial class BlockUpgradeState
 
         /// <summary>\[nextyes\] Number of blocks which approved the protocol upgrade.</summary>
         [Newtonsoft.Json.JsonProperty("next-protocol-approvals", Required = Newtonsoft.Json.Required.Default, NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
-        public int? NextProtocolApprovals { get; set; }
+        public long? NextProtocolApprovals { get; set; }
```
Comment (WesleyMiller1998) [inline dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:586]: I'm seeing that a lot of these have been changed to long, but I believe that all amounts, rounds, and asset ids should be an unsigned integer with base 64. That would mean that these should really be ulong

**Review by WesleyMiller1998 — CHANGES_REQUESTED** [review]
I believe that a lot of these should be ulong instead of long and there is a strange conditional

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs (hunk 3 above)
Comment (FrankSzendzielarz) [reply dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:586]: Agreed , long should be ulong for types derived from things used in TEAL. Let me try to figure out some way of determining what integers should be what and submit a PR for this.

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs [hunk 4]
```diff
@@ -58,7 +58,7 @@ public partial class Account
 
         /// <summary>\[teap\] the sum of all extra application program pages for this account.</summary>
         [Newtonsoft.Json.JsonProperty("apps-total-extra-pages", Required = Newtonsoft.Json.Required.Default, NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
-        public int? AppsTotalExtraPages { get; set; }
+        public long? AppsTotalExtraPages { get; set; }
```
Comment (FrankSzendzielarz) [inline dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:61]: Good example here - should int? really be long? for this??

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs (hunk 1 above)
Comment (FrankSzendzielarz) [reply dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:786]: Auto-generated. The JSON spec says that the fields are required:

`  "StateSchema": {
      "description": "Represents a \\[apls\\] local-state or \\[apgs\\] global-state schema. These schemas determine how much storage may be used in a local-state or global-state for an application. The more space used, the larger minimum balance must be maintained in the account holding the data.",
      "type": "object",
      "required": [
        "num-uint",
        "num-byte-slice"
      ],`

Comment (scholtz) [reply dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:786]: yeah, the schema might be wrong.. the reality is that it is failing parsing data

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs (hunk 3 above)
Comment (scholtz) [reply dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:586]: i agree

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs (hunk 4 above)
Comment (scholtz) [reply dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:61]: i agree.. we should find out somehow what is the real datatype in golang

File: dotnet-algorand-sdk/V2/Indexer/Model/Model.cs (hunk 1 above)
Comment (FrankSzendzielarz) [reply dotnet-algorand-sdk/V2/Indexer/Model/Model.cs:786]: I have found the go models and StateSchema properties should be non nullable but required.

**Pull request merged**